### PR TITLE
Implement a test suite for server, basic endpoint implementation.

### PR DESCRIPTION
**Pull request by SamPetcher**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+afterAll(() => pool.end());
+beforeEach(() => seed(testData));
+
+
+describe('GET /api/topics', () => {
+	test('Should return a list of topics in the form of an array', () => {
+		return request(app).get('/api/topics')
+			.expect(200)
+			.then( (r) => {
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: r isn't an acceptable variable name, can you make it more verbose please

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+				expect(Array.isArray(r.body)).toBe(true)
+				expect(r.body[0]).toEqual({slug: 'mitch', description: 'The man, the Mitch, the legend'})
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: checking that part of the object looks exactly how you expect with toEqual is very brittle, what would happen if we later patched this object? Can you refactor this using the jest matchers .toMatchObject or .objectContaining

File: models/news.models.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const db = require('../db/connection.js')
+
+
+exports.selectTopics = () => {
+    return db.query('SELECT * FROM topics;').then( (r) => r.rows )
```
Comment (roisinoreilly) [inline models/news.models.js]: again can we use a better variable name

File: controllers/news.controllers.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+const { selectTopics } = require("../models/news.models")
+
+exports.getTopics = (req, res, next) => selectTopics().then((r) => {
```
Comment (roisinoreilly) [inline controllers/news.controllers.js]: better variable name please

File: __tests__/app.test.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+			})
+			})
+	})
+
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: can you include the sad path in your testing here

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
A good start! Can you add in a test for the sad path and refactor how you're asserting an object should look, and get rid of all your instances of "r" to make your code more readable. I'd also suggest for future to split your commits into more than one if you have to use a comma or the word and in your commit message

File: __tests__/app.test.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+afterAll(() => pool.end());
+beforeEach(() => seed(testData));
+
+
+describe.only('GET /api/topics', () => {
+	test('Should return a list of topics in the form of an array', () => {
+		return request(app).get('/api/topics')
+			.expect(200)
+			.then( (topics) => {
+				expect(Array.isArray(topics.body)).toBe(true)
```
Comment (B7Bprog) [inline __tests__/app.test.js]: You are using .forEach array method so is checking for array necessary here? ^^

File: __tests__/app.test.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+afterAll(() => pool.end());
+beforeEach(() => seed(testData));
+
+
+describe.only('GET /api/topics', () => {
+	test('Should return a list of topics in the form of an array', () => {
+		return request(app).get('/api/topics')
+			.expect(200)
+			.then( (topics) => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:17]: Please check the length of the array as well.

File: __tests__/app.test.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+const seed = require('../db/seeds/seed.js');
+const pool = require('../db/connection.js');
+const { response } = require('../app');
+
+
+afterAll(() => pool.end());
+beforeEach(() => seed(testData));
+
+
+describe.only('GET /api/topics', () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:13]: Remove .only from here and  from all other places.

File: __tests__/app.test.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+					expect(topics).toEqual(expect.objectContaining({
+						slug: expect.any(String),
+						description: expect.any(String),
+					}))
+			})	
+			})
+	})
+})
+// We will assert with jest whether contains x y or Z
+describe('GET /api/articles', () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js]: This test doesn't run as you have a .only in your previous describe block.

File: __tests__/app.test.js [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+	test('Should return a list of articles', () => {
+		return request(app).get('/api/articles')
+			.expect(200)
+			.then( (reponse) => {
+				expect(Array.isArray(response.body)).toBe(true)
+				expect(Object.keys(response.body[0])).toEqual([`author`, `title`, `article_id`, `topic`, `created_at`, `votes`, `comment_count`])
```
Comment (B7Bprog) [inline __tests__/app.test.js]: Try to use .toMatchObject or .objectContaining to allow further expansion.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Samuel, you are on the right path, just need a few modifications. Please see my comments, refactor your code and please remove all '.only' from your tests and make sure everything passes.

File: __tests__/app.test.js (hunk 6 above)
Comment (SamPetcher) [reply __tests__/app.test.js]: My bad. From prior.

File: __tests__/app.test.js (hunk 9 above)
Comment (SamPetcher) [reply __tests__/app.test.js]: Ah. This is from a later checkpoint I was working on. It is an incomplete test. Is it possible for me to circumvent husky when I move onwards in this way?

File: __tests__/app.test.js (hunk 10 above)
Comment (SamPetcher) [reply __tests__/app.test.js]: As previous. It's a copy and paste job (my own code) that I hadn't finished implementing yet!

Comment (SamPetcher) [reply __tests__/app.test.js]: I get my mistake here now. This is on the wrong branch entirely.

**Pull request merged**